File: en Python/exercice4_3.py
import os
import sys

tube_l, tube_e = os.pipe()
pid = os.fork()
if pid == 0:
    os.close(tube_e)
    fd = str(tube_l)
    os.execlp("python3", "python3", "fils.py", fd)

else:
    os.close(tube_le)
    os.write(tube_e, b"Bonjour")

File: En C/exercice4_3.c
#include <stdio.h>
#include <unistd.h>
#include <string.h>
#include <sys/wait.h>
int main()
{
    int tube[2] ;
    pipe(tube) ;

    pid_t pid_fils = fork() ;
    if (pid_fils == 0)
    {
        char fd[10];
        sprintf(fd, "%d", tube[0]);
        char *arg[] = {"./fils", fd, NULL};
        execv("./fils", arg);
    }
    else
    {
        close(tube[0]);
        write(tube[1], "Bonjour", 8);
    }

    return (0);

}